### PR TITLE
Migrate CI to GHA

### PR DESCRIPTION
**Pull request by kokosing**
Migrate CI to GHA

### CODE REVIEW

**Comment by kokosing** [comment]
This is passing in my fork: https://github.com/kokosing/trino-maven-plugin/actions/runs/4054466677/jobs/6976331969

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Java CI
+
+on: [push]
```
Comment (findepi) [inline .github/workflows/ci.yaml:3]: `pull_request`?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
```
Comment (findepi) [inline .github/workflows/ci.yaml:15]: any particular reason for temurin?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots package
```
Comment (findepi) [inline .github/workflows/ci.yaml:17]: why `--update-snapshots`?
there should be no snapshots in local repo, so it is not useful for the CI

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: Java CI
```
Comment (findepi) [inline .github/workflows/ci.yaml:1]: "ci"

**Review by findepi — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
```
Comment (findepi) [inline .github/workflows/ci.yaml:11]: as a followup we should update https://github.com/trinodb/trino-maven-plugin/blob/master/pom.xml#L30 too

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (kokosing) [reply .github/workflows/ci.yaml:15]: No. Changed to `zulu`.

File: .github/workflows/ci.yaml (hunk 5 above)
Comment (kokosing) [reply .github/workflows/ci.yaml:11]: Yes.

**Comment by kokosing** [comment]
Tested with https://github.com/kokosing/trino-maven-plugin/pull/1

**Pull request merged**